src/QsCompiler/CompilationManager/TypeChecking.cs

```diff
@@ -2015,6 +2015,13 @@ QsCallable GetCallable((QsQualifiedName, ImmutableArray<QsSpecialization>) specI
                 }
                 compilation.UpdateCallables(callables);
                 compilation.UpdateTypes(types);
+                var callGraph = Transformations.CallGraphWalker.BuildCallGraph.Apply(callables);
+                foreach (var (parent, diag) in callGraph.VerifyAllCycles())
+                {
+                    var info = callableDeclarations[parent];
+                    var offset = info.Position is DeclarationHeader.Offset.Defined pos ? pos.Item : null;
+                    diagnostics.Add(Diagnostics.Generate(info.SourceFile.Value, diag, offset));
```

Comment:
I was testing this with the language server and the VS Code plugin, and I'm not sure this is sufficient to keep the diagnostics up-to-date. I was able to come up with this example:

```
    operation Foo<'a>() : Unit {
        Bar<'a[]>();
    }

    operation Bar<'a>() : Unit {
        Message("bar");
        // Foo<'a>();
    }
```

where if I uncommented `// Foo<'a>();`, it would not give a diagnostic for a bad cycle until I manually saved the file. Similarly, if I commented that line again, the error message on that line would go away, but the error message at the `Bar` call would stay until I saved the file.

Reply:
A related issue: if `Foo<'a>();` is uncommented, and I edit the source code of `Bar` without changing the call to `Foo`, the diagnostic for the `Foo` call will disappear.